src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php

```diff
@@ -865,6 +868,7 @@ public function testTypedProperties()
         $this->assertEquals(Type::list(Type::string()), $this->extractor->getType(Php74Dummy::class, 'stringCollection'));
         $this->assertEquals(Type::nullable(Type::int()), $this->extractor->getType(Php74Dummy::class, 'nullableWithDefault'));
         $this->assertEquals(Type::array(), $this->extractor->getType(Php74Dummy::class, 'collection'));
+        $this->assertEquals(Type::nullable(Type::list(Type::object(Dummy::class))), $this->extractor->getType(Php74Dummy::class, 'nullableTypedCollection'));
```

Comment:
we should expand `testTypedPropertiesLegacy()` in the same way to ensure that the behaviour of old and new type system matches